src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/csip/CSIPService.kt

```diff
@@ -1,22 +1,83 @@
 package uk.gov.justice.digital.hmpps.prisonertonomisupdate.csip
 
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
 import com.microsoft.applicationinsights.TelemetryClient
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
 import uk.gov.justice.digital.hmpps.prisonertonomisupdate.config.trackEvent
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.nomismappings.model.CSIPFullMappingDto
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.services.CreateMappingRetryMessage
 import uk.gov.justice.digital.hmpps.prisonertonomisupdate.services.CreateMappingRetryable
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.services.synchronise
 
 @Service
 class CSIPService(
   private val telemetryClient: TelemetryClient,
   private val nomisApiService: CSIPNomisApiService,
+  private val dpsApiService: CSIPDpsApiService,
   private val mappingApiService: CSIPMappingApiService,
+  private val csipRetryQueueService: CSIPRetryQueueService,
+  private val objectMapper: ObjectMapper,
 ) : CreateMappingRetryable {
   private companion object {
     val log: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
+  suspend fun createCSIPReport(csipEvent: CSIPEvent) {
+    val dpsCsipReportId = csipEvent.additionalInformation.recordUuid
+    val offenderNo = requireNotNull(csipEvent.personReference.findNomsNumber())
+    val telemetryMap = mapOf(
+      "dpsCsipReportId" to dpsCsipReportId,
+      "offenderNo" to offenderNo,
+    )
+
+    if (csipEvent.wasCreatedInDPS()) {
+      synchronise {
+        name = "csip"
+        telemetryClient = this@CSIPService.telemetryClient
+        retryQueueService = csipRetryQueueService
+        eventTelemetry = telemetryMap
+
+        checkMappingDoesNotExist {
+          mappingApiService.getOrNullByDpsId(dpsCsipReportId)
+        }
+        transform {
+          dpsApiService.getCsipReport(dpsCsipReportId)
+            .let { dpsCsip ->
+              nomisApiService.upsertCsipReport(dpsCsip.toNomisUpsertRequest()).let { nomisCsip ->
+                CSIPFullMappingDto(
+                  dpsCSIPReportId = dpsCsipReportId,
+                  nomisCSIPReportId = nomisCsip.nomisCSIPReportId,
+                  mappingType = CSIPFullMappingDto.MappingType.DPS_CREATED,
+                  attendeeMappings = listOf(),
+                  factorMappings = listOf(),
+                  interviewMappings = listOf(),
+                  planMappings = listOf(),
+                  reviewMappings = listOf(),
+                )
+              }
+            }
+        }
+        saveMapping { mappingApiService.createMapping(it) }
```

Comment:
More work still to be done here to cater for child mappings - but this is the most basic insert